src/assets/VueOptionVsCompositionAPI/option-vs-composition-api.md

```diff
@@ -0,0 +1,186 @@
+
+The overall script snippet code looks like this at last for the composition API:
+```js
+import { reactive, toRefs, computed , watch} from "vue";
+export default {
+  setup() {
+    // data
+    const data = reactive({
+      name: 'Sagar Gurung',
+      age: 26,
+      votingStatus: "Not Eligible",
+      bornYear: 1997
+    })
+    const mainData = toRefs(data)
+    // methods
+    function increaseAge() {
+      data.age++
+    }
+    function decreaseAge() {
+      data.age--
+    }
+    //computed
+    const getVotingStatus = computed(() => {
+      return (data.age >= 18) ? 'Eligible' : 'Not Eligible'
+    })
+    //watcher
+    watch(mainData.age, (oldAge, newAge) => {
+      if(oldAge > newAge) {
+        data.bornYear--
+      } else {
+        data.bornYear++
+      }
+    })
+
+    return {
+      ...mainData,
+      increaseAge,
+      decreaseAge,
+      getVotingStatus
+    }
+  }
+}
+```
+In the overall code, we have imported `reactive`, `toRefs`, `computed`, and `watch` needed for building components with Composition API. The `toRefs` is used to convert the data created with `reactive` to plain objects. Also, In Composition API we need to define all those `data`, `computed`, `watch`, and `methods` inside of it as done above.
+And at last, all the thing that has been defined `data`, `computed`, and `methods` needs to be returned to be exposed to the template.
+
+In this way, we can simply convert a component defined with `Option API` to `Composition API`. But This is just a basic example. There are lots of things that we can perform using composition API.
+I hope this blog at least gave you some insight on `Composition API` and also how you can replace your component defined in `Option API` with it. But before finishing the blog I would like to highlight some
+points regarding both `Option API` and `Composition API` so that you can relate which one to prefer before building up your projects.
+
+## Option API and Composition API
+
+You can find lots of points regarding the comparison among these two APIs. Still, I would only like to highlight some of it as both approaches for building up components have their benefits and drawbacks.
+
+1. Composition API is new to Vue which can be called a modern way to define components whereas on the other hand Option API is an old or traditional way to build up components in Vue.
+2. Also, the structure and syntax for the Options API seem a bit simpler and easier to understand than that of the Composition API. It's a good choice for small projects and also for beginners. But for large and complex projects a better choice would be Composition API since it is a more flexible and modular structure that enhances better code maintainability.
+3. In addition to this since Composition API is fully based on functional, reactive programming it can be difficult for learners who are not familiar with such approach. In this case, Option API is a more captivating choice.
```

Comment:
```suggestion
3. In addition to this, since Composition API is fully based on functional, reactive programming it can be difficult for learners who are not familiar with such approach. In this case, Option API is a more captivating choice.
```